clang/test/SemaCXX/array-bounds.cpp

```diff
@@ -170,16 +170,14 @@ void test_nested_switch() {
   }
 }
 
-// Test that if all the values of an enum covered, that the 'default' branch
-// is unreachable.
+// Test that a warning is not emitted if the code is unreachable.
 enum Values { A, B, C, D };
 void test_all_enums_covered(enum Values v) {
```

Comment:
I think what this tests is checking is that we don’t emit a warning for the invalid array access because it is supposed to be unreachable. With this pr, this is no longer treated as unreachable, so the right change here imo would be to check that the warning *is* emitted.

Reply:
Thanks for taking a look. What I was trying to do here was make a control flow graph where the out of bounds access happened in a path clang knew was unreachable (and thus not fire). I'm not sure if that's tested anywhere else. Checking if the warning is emitted here with the switch seems a little redundant to me with other checks about no longer assuming the default path is unreachable if enum values are covered. Thoughts?

Reply:
I mean, I’d imagine we have tests for that elsewhere, if not, then
```c++
void f() {
    int x[2];
    goto foo;
    x[2] = 7;
    foo:;
}
```
works. If that’s all we care about here, then we can just replace it with this, otherwise, we can probably just remove it entirely. I haven’t looked at the rest of this file too much.

Reply:
It looks like there's another test case that does effectively make an unreachable branch, I'll update this case as you suggested